docs/codesigning/common-issues.md

```diff
@@ -32,8 +32,121 @@ Your project defines a provisioning profile which doesn't exist on your local ma
 
 Make sure to have a valid code signing identity defined in your project targets. This might happen when you select `Don't Code Sign` as Code Signing Identity.
 
-## fastlane errors
+## _fastlane_ errors
 
 > Could not find a matching code signing identity for type 'X'
 
 There are no certificates available on the Apple Developer Portal. This could either mean someone revoked a certificate, or you don't have access to it. 
+
+# Debugging Steps for "No applicable devices found." error
+
+It's not immediately obvious what this error is or what it's caused by, so hopefully these simple steps will help you find the root cause faster than my experience with it.
+
+### Not Just _fastlane_
+
+This error happens to all projects built with `xcodebuild` / `xcrun`, so if you're trying to blame _fastlane_ here, take a deep breath and look elsewhere first. _fastlane_ might help alleviate it with some env var unsetting / setting or other tricks (see below), but this is an error from the new export APIs that Xcode uses.
+
+### Legacy Export API
+
+As of Xcode 8.3, the deprecated / legacy export api (e.g. `use_legacy_build_api`) is removed so you're unfortunately stuck with using the latest, and often broken, export API.
+
+### Red Herring
+
+The error message, `"No applicable devices found."`, is actually just a red herring: it's not the real error. Something else is erroring out under the hood so you'll need to dig in to find out what.
```

Comment:
What's a `red herring`, maybe let's use a phrase that's more commonly known for non-native speakers